src/main/java/org/spongepowered/common/mixin/api/mcp/entity/EntityMixin_API.java

```diff
@@ -682,4 +684,32 @@ public EntityArchetype createArchetype() {
         this.get(VelocityData.class).ifPresent(manipulators::add);
     }
 
+    @Override
+    public Set<String> getTags() {
+        return Collections.unmodifiableSet(this.tags);
+    }
+
+    @Override
+    public boolean addTag(final String tag) {
```

Comment:
I think this would be better off being `@Intrinsic` and calling through to the original method so it doesn't interfere with other vanilla calls.

```
@Intrinsic
public boolean entity$addTag(final String tag) {
    Objects.requireNonNull(tag, "tag");
    return this.shadow$addTag(tag);
}
```

The difference between this method and that one is that you return the result of `Set#add`, vanilla just returns `true` - is it a problem if `addTag` just reports whether the entry is in the set, which it will be if it reports true, regardless?

Reply:
My original intent on this was to prevent developers to put a null value in the set.
But I agree we can report true as vanilla does